overlord/snapshotstate/snapshotmgr.go

```diff
@@ -129,6 +194,13 @@ func prepareSave(task *state.Task) (snapshot *snapshotSetup, cur *snap.Info, cfg
 		}
 	}
 
+	// this should be done last because of it modifies the state and the caller needs to undo this if other operation fails.
+	if snapshot.Auto {
+		if err := saveExpiration(st, snapshot.SetID, time.Now().Add(automaticSnapshotExpiration(st))); err != nil {
```

Comment:
this code path doesn't seem to be tested,  probably also means the corresponding undo code in doSave is tested only as no-op

Reply:
Test added.